### PR TITLE
[Merged by Bors] - Feat: port Data.Set.Countable

### PR DESCRIPTION
**Pull request by urkud**
---

- [x] depends on: #1787

Force-pushed to cleanup git history from commits that belong to other PRs.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#1787~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by jcommelin** [comment]
It looks like the `Countable` typeclass is not `#align`ed. Some names in comments should be fixed.

File: Mathlib/Data/Set/Countable.lean [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+#align set.countable_iff_exists_surjective Set.countable_iff_exists_surjective
+
+alias Set.countable_iff_exists_surjective ↔ Countable.exists_surjective _
+#align set.countable.exists_surjective Set.Countable.exists_surjective
+
+theorem countable_univ [Countable α] : (univ : Set α).Countable :=
+  to_countable univ
+#align set.countable_univ Set.countable_univ
+
+/-- If `s : set α` is a nonempty countable set, then there exists a map
```
Comment (jcommelin) [inline Mathlib/Data/Set/Countable.lean]:
```suggestion
/-- If `s : Set α` is a nonempty countable set, then there exists a map
```

**Comment by urkud** [comment]
Aligned, fixed some comments.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**